src/kemal/route_handler.cr

```diff
@@ -33,7 +34,11 @@ module Kemal
 
       route = @routes.find(lookup_path)
 
-      @cached_routes[lookup_path] = route if route.found?
+      if route.found? && @cached_routes.size < CACHED_ROUTES_LIMIT
+        @cached_routes[lookup_path] = route
+      else
```

Comment:
This clears the cache every time a path doesn't resolve to a route. Not very effective.